salt/grains/smartos.py

```diff
@@ -139,30 +139,31 @@ def _smartos_zone_pkgsrc_data():
         'pkgsrcpath': 'Unknown',
     }
 
-    pkgsrcversion = re.compile('^release:\\s(.+)')
-    if os.path.isfile('/etc/pkgsrc_version'):
-        with salt.utils.files.fopen('/etc/pkgsrc_version', 'r') as fp_:
-            for line in fp_:
-                line = salt.utils.stringutils.to_unicode(line)
-                match = pkgsrcversion.match(line)
-                if match:
-                    grains['pkgsrcversion'] = match.group(1)
-
     pkgsrcpath = re.compile('PKG_PATH=(.+)')
-    if os.path.isfile('/opt/local/etc/pkg_install.conf'):
-        with salt.utils.files.fopen('/opt/local/etc/pkg_install.conf', 'r') as fp_:
-            for line in fp_:
-                line = salt.utils.stringutils.to_unicode(line)
-                match = pkgsrcpath.match(line)
-                if match:
-                    grains['pkgsrcpath'] = match.group(1)
+    pkgsrcversion = re.compile('^https?://pkgsrc.joyent.com/packages/SmartOS/(.+)/(.+)/All$')
```

Comment:
Is this the only URL this will ever be? Also, is there a better place to put this - maybe as a "constant" at the toplevel of the file? I'm just trying to think of where would be the best place to put this in the case that they change up their URL schemes.

Reply:
On SmartOS (which these grains are specific for) this should be the only URL we'll encounter. The URL is probably different on NetBSD which also uses pkgsrc, but I'm not sure they even use the same release tagging methode that SmartOS has.

Reply:
I have added a note above that explains why 'Unknown' would the the correct value of a sysadmin decided to create there own 'spin' of a SmartOS native zone. (Sort of like creating an entirely new linux distribution. e.g. debian <-> redhat <-> suse)

So it is clear why there is only one URL provided from the code + comment.